crates/astria-conductor/src/celestia/verify.rs

```diff
@@ -94,24 +98,37 @@ struct VerificationTaskKey {
 pub(super) async fn verify_metadata(
     blob_verifier: Arc<BlobVerifier>,
     converted_blobs: ConvertedBlobs,
+    mut executor: executor::Handle<StateIsInit>,
 ) -> VerifiedBlobs {
     let (celestia_height, header_blobs, rollup_blobs) = converted_blobs.into_parts();
 
     let mut verification_tasks = JoinMap::new();
     let mut verified_header_blobs = HashMap::with_capacity(header_blobs.len());
 
+    let next_expected_firm_sequencer_height =
+        executor.next_expected_firm_sequencer_height().value();
+
     for (index, blob) in header_blobs.into_iter().enumerate() {
-        verification_tasks.spawn(
-            VerificationTaskKey {
-                index,
-                block_hash: blob.block_hash(),
-                sequencer_height: blob.height(),
-            },
-            blob_verifier
-                .clone()
-                .verify_metadata(blob)
-                .in_current_span(),
-        );
+        if blob.height().value() < next_expected_firm_sequencer_height {
+            info!(
+                next_expected_firm_sequencer_height,
+                sequencer_height_in_metadata = blob.height().value(),
+                "dropping Sequencer metadata item without verifying against Sequencer because its \
+                 height is below the next expected firm height"
+            );
```

Comment:
is this guaranteed safe? ie are these blobs already guaranteed to be executed/verified as soft blocks already and we're sure that they are the same as the ones already verified?

Reply:
Note that this is checking against the next expected *firm* height - this is independent of the soft blocks.